packages/react-components/react-motion-components-preview/library/src/components/Collapse/collapse-atoms.ts

```diff
@@ -0,0 +1,155 @@
+import { AtomMotion } from '@fluentui/react-motion/src/types';
+import type { CollapseOrientation } from './collapse-types';
+
+// ----- SIZE -----
+
+const sizeValuesForOrientation = (orientation: CollapseOrientation, element: Element) => {
```

Comment:
All of this happens in runtime and has some cost. I think that it would be nice to reduce computations there. For example, avoid repeated calls of `sizeValuesForOrientation()` & `whitespaceValuesForOrientation()`

Reply:
It's a few ternaries that execute when a motion is defined. The cost is negligible, the repetition is negligible, and I'm not sure where else the conditional code could be shunted to.